### PR TITLE
Separate s3 and nfs test

### PR DESCRIPTION
**Pull request by khushboo-rancher**
https://github.com/longhorn/longhorn/issues/2999

Signed-off-by: Khushboo <fnu.khushboo@suse.com>

### CODE REVIEW

**Review by innobead — CHANGES_REQUESTED** [review]
This is an easy way to separate the test execution w/ different backup stores. 

But I would expect if we can leverage the same scripts, Jenkinsfile, manifests, etc w/ some conditional check inside to have an adaptive run when giving different test runtime configurations. This will be much better for test setup management. WDYT?

**Comment by khushboo-rancher** [comment]
@innobead I have made the changes as per suggestion. Could you please have a look.
cc: @meldafrawi

File: test_framework/scripts/longhorn-setup.sh [hunk 1]
```diff
@@ -163,7 +163,16 @@ run_longhorn_upgrade_test(){
...
-	LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test.yaml"
+	if [[ $BACKUP_STORE_TYPE = "nfs" ]]; 
+		then
+			LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test_nfs.yaml"
+		elif [[ $BACKUP_STORE_TYPE = "s3" ]];
+			then
+				LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test_s3.yaml"
+		else
+			LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test.yaml"
+	fi
```
Comment (innobead) [inline test_framework/scripts/longhorn-setup.sh:174]:
```suggestion
	if [[ $BACKUP_STORE_TYPE = "nfs" ]]; then
	  LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test_nfs.yaml"
	elif [[ $BACKUP_STORE_TYPE = "s3" ]]; then
	  LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test_s3.yaml"
	else
			LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test.yaml"
	fi
```

File: test_framework/scripts/longhorn-setup.sh [hunk 2]
```diff
@@ -209,7 +218,16 @@ run_longhorn_tests(){
...
+
+	if [[ $BACKUP_STORE_TYPE = "nfs" ]]; 
+		then
+			LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test_nfs.yaml"
+		elif [[ $BACKUP_STORE_TYPE = "s3" ]];
+			then
+				LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test_s3.yaml"
+		else
+			LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test.yaml"
+	fi
```
Comment (innobead) [inline test_framework/scripts/longhorn-setup.sh:230]:
```suggestion
	if [[ $BACKUP_STORE_TYPE = "nfs" ]]; then
	  LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test_nfs.yaml"
	elif [[ $BACKUP_STORE_TYPE = "s3" ]]; then
		LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test_s3.yaml"
	else
	  LONGHORN_TESTS_MANIFEST_FILE_PATH="${LONGHORH_TESTS_REPO_BASEDIR}/manager/integration/deploy/test.yaml"
	fi
```

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just a format suggestion to resolve.

**Review by innobead — APPROVED** [review]
LGTM

**Pull request merged**